src/SFA.DAS.Assessor.Functions/Functions/Ofqual/OfqualDownloader.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.Assessor.Functions.Domain.Entities.Ofqual;
+using SFA.DAS.Assessor.Functions.Domain.OfqualImport.Interfaces;
+using SFA.DAS.Assessor.Functions.Infrastructure;
+
+namespace SFA.DAS.Assessor.Functions.Functions.Ofqual
+{
+    internal abstract class OfqualDownloader
+    {
+        private readonly IOfqualDownloadsBlobFileTransferClient _blobFileTransferClient;
+        private readonly HttpClient _httpClient;
+        private readonly OfqualDataType _ofqualFileType;
+        private readonly string _ofqualDataTypeKeyword;
+
+        protected OfqualDownloader(
+            IOfqualDownloadsBlobFileTransferClient blobFileTransferClient,
+            HttpClient httpClient,
+            OfqualDataType ofqualFileType)
+        {
+            _blobFileTransferClient = blobFileTransferClient;
+            _httpClient = httpClient;
+            _ofqualFileType = ofqualFileType;
+            _ofqualDataTypeKeyword = GetOfqualDataTypeKeyword(ofqualFileType);
+        }
+
+        protected async Task<string> DownloadData(ILogger logger)
+        {
+            logger.LogInformation("Downloading {datatype} data from {address}.", _ofqualDataTypeKeyword, _httpClient.BaseAddress);
+
+            try
+            {
+                using HttpResponseMessage response = await _httpClient.GetAsync(_httpClient.BaseAddress);
+                response.EnsureSuccessStatusCode();
+                string responseBody = await response.Content.ReadAsStringAsync();
+
+                logger.LogInformation("Finished downloading {datatype} data.", _ofqualDataTypeKeyword);
+
+                return await SaveDownloadedFile(responseBody, logger);
+            }
+            catch (HttpRequestException ex)
+            {
+                logger.LogError(ex, "Error: {Message}", ex.Message);
+                throw;
+            }
+        }
+
+        private async Task<string> SaveDownloadedFile(string content, ILogger logger)
+        {
+            string filename = OfqualDataFileName.CreateForFileType(_ofqualFileType);
+            string filePath = $"Downloads/{filename}";
+
+            await _blobFileTransferClient.UploadFile(content, filePath);
+            logger.LogInformation("File saved at {filePath}", filePath);
+
+            return filePath;
+        }
+
+        private static string GetOfqualDataTypeKeyword(OfqualDataType ofqualFileType)
+        {
+            return ofqualFileType switch
+            {
+                OfqualDataType.Organisations => nameof(OfqualDataType.Organisations),
+                OfqualDataType.Qualifications => nameof(OfqualDataType.Qualifications),
+                _ => throw new ArgumentException($"{ofqualFileType} is an unknown Ofqual data type.")
+            };
+        }
```

Comment:
This method isn't necessary -- its functionality can be replaced by calling `ToString()` on the `OfqualDataType` enum itself.